### DISCUSSION TITLE
Grafana stops working when i add the 2 lines in yml

### DISCUSSION THREAD

**Discussion by djpalmis** [jheredianet/Teslamate-CustomGrafanaDashboards#10] [Q&A] [answered]
![Screenshot_20230115_215146_Samsung Internet](https://user-images.githubusercontent.com/402122/212568039-7a795515-5a08-46c5-ab7a-500b62337aed.jpg)
When i surf my website/grafana it says not found.
Removing the 2 lines it works again.

Any tips?
I use teslamate on google cloud with https.

**Comment by jheredianet** [comment]
Could you show you complete Grafana service definition, and log?
Remove any sensitive information. 
I guess you have already clone the repo first. Also have in mind you folder structure.

To get Grafana log, you should do something like:

`docker-compose logs -f grafana`

**Comment by djpalmis** [comment]
yes i cloned it.
my folder structure:
![Screenshot_20230115_222953_Samsung Internet](https://user-images.githubusercontent.com/402122/212568364-39988508-d530-4ac0-b063-b18594ee4ed0.jpg)
about the service definition i guess you mean the following:
![Screenshot_20230115_223104_Samsung Internet](https://user-images.githubusercontent.com/402122/212568482-46257ecf-713b-4600-9a5c-24cd69d0ecc0.jpg)

![Screenshot_20230115_223116_Samsung Internet](https://user-images.githubusercontent.com/402122/212568488-6fca9c73-1b1c-48fa-a4ae-73e7d8202232.jpg)

i try to get the log now

**Reply by jheredianet** [reply]
Every seems to be ok, let's see the log to understand what is going on with Grafana

**Reply by SiNaPsEr0x** [reply]
```
logger=ngalert.multiorg.alertmanager t=2023-01-15T21:34:33.72+0000 lvl=info msg="starting MultiOrg Alertmanager"
logger=provisioning t=2023-01-15T21:34:33.72+0000 lvl=eror msg="Failed to provision dashboard" error="Failed to create provisioner: Failed to read dashboards config: could not parse provisioning config file: customdashboards.yml error: read /etc/grafana/provisioning/dashboards/customdashboards.yml: is a directory"
logger=server t=2023-01-15T21:34:33.72+0000 lvl=eror msg="Stopped background service" service=*provisioning.ProvisioningServiceImpl reason="Failed to create provisioner: Failed to read dashboards config: could not parse provisioning config file: customdashboards.yml error: read /etc/grafana/provisioning/dashboards/customdashboards.yml: is a directory"
logger=ngalert t=2023-01-15T21:34:33.72+0000 lvl=eror msg="unable to fetch orgIds" msg="context canceled"
logger=http.server t=2023-01-15T21:34:33.72+0000 lvl=info msg="HTTP Server Listen" address=[::]:3000 protocol=http subUrl=/grafana socket=
*provisioning.ProvisioningServiceImpl run error: Failed to create provisioner: Failed to read dashboards config: could not parse provisioning config file: customdashboards.yml error: read /etc/grafana/provisioning/dashboards/customdashboards.yml: is a directory
logger=server t=2023-01-15T21:34:33.72+0000 lvl=eror msg="Server shutdown" error="*provisioning.ProvisioningServiceImpl run error: Failed to create provisioner: Failed to read dashboards config: could not parse provisioning config file: customdashboards.yml error: read /etc/grafana/provisioning/dashboards/customdashboards.yml: is a directory"
```

**Reply by SiNaPsEr0x** [reply]
some error on google

**Reply by djpalmis** [reply]
the part of my log that i think shall help
![Screenshot_20230115_224821_Samsung Internet](https://user-images.githubusercontent.com/402122/212569019-ab37e387-c7a5-46cc-96ea-bb8a7a553bb2.jpg)

**Comment by jheredianet** [comment, marked as answer]
The error seems to be here

```
/etc/grafana/provisioning/dashboards/customdashboards.yml: is a directory
```
It may be, because you got a previous error on cloning. 

Try this (to begin again):

```
rm -rf Teslamate-CustomGrafanaDashboards/
docker-compose down
git clone https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards.git
docker-compose up -d
```

**Reply by SiNaPsEr0x** [reply]
some error...

with traefik the port:3000 is relink to http://path/grafana... mmm this problem?

```
logger=migrator t=2023-01-15T21:47:39.47+0000 lvl=info msg="migrations completed" performed=0 skipped=392 duration=907.721µs
logger=plugin.manager t=2023-01-15T21:47:39.62+0000 lvl=info msg="Plugin registered" pluginId=input
logger=plugin.signature.validator t=2023-01-15T21:47:39.96+0000 lvl=warn msg="Permitting unsigned plugin. This is not recommended" pluginID=panodata-map-panel pluginDir=/var/lib/grafana-plugins/grafana-worldmap-panel-ng
logger=plugin.manager t=2023-01-15T21:47:39.96+0000 lvl=info msg="Plugin registered" pluginId=panodata-map-panel
logger=plugin.manager t=2023-01-15T21:47:39.96+0000 lvl=info msg="Plugin registered" pluginId=natel-plotly-panel
logger=plugin.manager t=2023-01-15T21:47:39.96+0000 lvl=info msg="Plugin registered" pluginId=pr0ps-trackmap-panel
logger=query_data t=2023-01-15T21:47:39.96+0000 lvl=info msg="Query Service initialization"
logger=live.push_http t=2023-01-15T21:47:39.97+0000 lvl=info msg="Live Push Gateway initialization"
logger=provisioning t=2023-01-15T21:47:40.06+0000 lvl=eror msg="Failed to provision dashboard" error="Failed to create provisioner: Failed to read dashboards config: could not parse provisioning config file: customdashboards.yml error: read /etc/grafana/provisioning/dashboards/customdashboards.yml: is a directory"
logger=server t=2023-01-15T21:47:40.06+0000 lvl=eror msg="Stopped background service" service=*provisioning.ProvisioningServiceImpl reason="Failed to create provisioner: Failed to read dashboards config: could not parse provisioning config file: customdashboards.yml error: read /etc/grafana/provisioning/dashboards/customdashboards.yml: is a directory"
logger=http.server t=2023-01-15T21:47:40.06+0000 lvl=info msg="HTTP Server Listen" address=[::]:3000 protocol=http subUrl=/grafana socket=
logger=server t=2023-01-15T21:47:40.06+0000 lvl=eror msg="Server shutdown" error="*provisioning.ProvisioningServiceImpl run error: Failed to create provisioner: Failed to read dashboards config: could not parse provisioning config file: customdashboards.yml error: read /etc/grafana/provisioning/dashboards/customdashboards.yml: is a directory"
*provisioning.ProvisioningServiceImpl run error: Failed to create provisioner: Failed to read dashboards config: could not parse provisioning config file: customdashboards.yml error: read /etc/grafana/provisioning/dashboards/customdashboards.yml: is a directory
```

**Reply by SiNaPsEr0x** [reply]
this is the folder... 

<img width="729" alt="image" src="https://user-images.githubusercontent.com/5438823/212569116-18a38d09-9cac-4333-8ba4-e72d51d3f280.png">


<img width="702" alt="image" src="https://user-images.githubusercontent.com/5438823/212569139-48acfd29-ad64-4b63-b1e7-c24a9e3d9d97.png">

**Reply by jheredianet** [reply]
No the error is not Traefik, if you see the last line of the logs it's complaining about the file customdashboards.yml
Could you verify if you have that file in the "dashboards" folder where you cloned the repo?

**Reply by SiNaPsEr0x** [reply]
<img width="845" alt="image" src="https://user-images.githubusercontent.com/5438823/212569195-ff209b9f-36b7-41a2-a214-79dc47155c1e.png">

**Reply by SiNaPsEr0x** [reply]
on my qnap NAS with :3000 port run correctly..

```
logger=plugin.manager t=2023-01-15T21:58:45.97+0000 lvl=info msg="Plugin registered" pluginId=pr0ps-trackmap-panel
logger=plugin.manager t=2023-01-15T21:58:45.97+0000 lvl=info msg="Plugin registered" pluginId=panodata-map-panel
logger=query_data t=2023-01-15T21:58:46.11+0000 lvl=info msg="Query Service initialization"
logger=live.push_http t=2023-01-15T21:58:46.13+0000 lvl=info msg="Live Push Gateway initialization"
logger=ngalert t=2023-01-15T21:58:47.03+0000 lvl=info msg="warming cache for startup"
logger=ngalert.multiorg.alertmanager t=2023-01-15T21:58:47.03+0000 lvl=info msg="starting MultiOrg Alertmanager"
logger=grafanaStorageLogger t=2023-01-15T21:58:47.04+0000 lvl=info msg="storage starting"
logger=http.server t=2023-01-15T21:58:47.07+0000 lvl=info msg="HTTP Server Listen" address=[::]:3000 protocol=http subUrl= socket=
logger=context traceID=00000000000000000000000000000000 userId=2 orgId=1 uname=lettura t=2023-01-15T21:58:59.7+0000 lvl=info msg="Request Completed" method=GET path=/api/live/ws status=0 remote_addr=192.168.1.18 time_ms=135 duration=135.50944ms size=0 referer= traceID=00000000000000000000000000000000
logger=live t=2023-01-15T21:59:13.46+0000 lvl=info msg="Initialized channel handler" channel=grafana/dashboard/uid/jchmNjtMTFx address=grafana/dashboard/uid/jchmNjtMTFx
[/share/Container/container-station-data/application/teslamate] # 

```

**Reply by jheredianet** [reply]
So estrange, it should work.
Could you comment the first line? It's where the problem is. It will let Grafana to work (but nothing will be imported, it's just a test)
It looks like docker is miss understanding that "customdashboards.yml" is a folder, not a file. 
```
      # - ~/Teslamate-CustomGrafanaDashboards/customdashboards.yml:/etc/grafana/provisioning/dashboards/customdashboards.yml
      - ~/Teslamate-CustomGrafanaDashboards/dashboards:/TeslamateCustomDashboards
```
I believe that maybe you docker is running with other user different to "tomcattopgun" and its reading other different home folder.

**Reply by SiNaPsEr0x** [reply]
yes... on GCC (google) docker run only with root... try it 1 moment

**Reply by jheredianet** [reply]
So, if it's running with root, you must clone with root or manually change the clone folder instead of (~/) you should put the full path

**Reply by SiNaPsEr0x** [reply]
ok runnly correctly but:

<img width="533" alt="image" src="https://user-images.githubusercontent.com/5438823/212569580-c142c370-9010-434d-a1cc-d3e8b25e2e12.png">

**Reply by SiNaPsEr0x** [reply]
> So, if it's running with root, you must clone with root or manually change the clone folder instead of (~/) you should put the full path

I've already tried but nothing changed becouse root have a privilege everywere

**Reply by SiNaPsEr0x** [reply]
some error with git clone in root mode

**Reply by jheredianet** [reply]
Yes, you should specify the full path on your docker compose where you cloned the repo, in stead of ~/ use the full path
```
      - /some/path/Teslamate-CustomGrafanaDashboards/customdashboards.yml:/etc/grafana/provisioning/dashboards/customdashboards.yml
      - /some/path/Teslamate-CustomGrafanaDashboards/dashboards:/TeslamateCustomDashboards
```

**Reply by SiNaPsEr0x** [reply]
yesssssssss

<img width="1222" alt="image" src="https://user-images.githubusercontent.com/5438823/212569785-e5bff9a0-3fbd-4456-91d4-dee83e5dd34d.png">


on google GCC this is fix


<img width="462" alt="image" src="https://user-images.githubusercontent.com/5438823/212569795-062db315-ea4a-449e-b52a-64efb972b6a7.png">

**Reply by jheredianet** [reply]
Good job!!!

**Reply by djpalmis** [reply]
thank You all guys! It works now!
